packages/core/src/html/linkProcessor.js

```diff
@@ -87,18 +87,28 @@ function convertMdAndMbdExtToHtmlExt(node) {
     if (hasNoConvert) {
       return;
     }
+
     const { href } = node.attribs;
-    const hasMdExtension = href.slice(-3) === '.md';
-    if (hasMdExtension) {
-      const newHref = `${href.substring(0, href.length - 3)}.html`;
-      node.attribs.href = newHref;
+
+    const hrefUrl = url.parse(href);
+
+    // get the first instance of URL fragment (first encounter of hash)
+    const urlFragment = hrefUrl.hash === null ? '' : hrefUrl.hash;
+    const urlFragmentIdx = urlFragment === '' ? href.length : href.indexOf(urlFragment);
+
+    let ext = '';
+    ext = href.substring(urlFragmentIdx - 3, urlFragmentIdx) === '.md' ? '.md' : ext;
```

Comment:
`href.substring(urlFragmentIdx - 3, urlFragmentIdx)` is available under `hrefUrl.pathname`

Reply:
Hmm actually I'm trying to get the extension here (not the pathname). In the updated implementation I used `path.posix.extname` on `hrefUrl.pathname` instead, thanks for your suggestion! :-)